File: lib/features/admin/domain/entities/group.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'group.freezed.dart';
part 'group.g.dart';

@freezed
class Group with _$Group {
  const factory Group({
    String? id,
    required String groupCode,
    String? title,
    String? description,
    int? course,
    String? courseId,
    required int shift,
    @Default(true) bool isActive,
  }) = _Group;

  factory Group.fromJson(Map<String, dynamic> json) => _$GroupFromJson(json);
}

File: server/models/Group.js
const mongoose = require('mongoose');

const GroupSchema = new mongoose.Schema({
    groupCode: {
        type: String,
        required: [true, 'Group code is required'],
        unique: true,
        trim: true
    },
    title: {
        type: String,
        trim: true
    },
    description: {
        type: String,
        trim: true
    },
    courseId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Course'
    },
    course: {
        type: Number,
        min: 1,
        max: 6
    },
    shift: {
        type: Number,
        required: [true, 'Shift is required'],
        enum: [1, 2],
        default: 1
    },
    isActive: {
        type: Boolean,
        default: true
    }
}, {
    timestamps: true
});

// Index for faster queries
module.exports = mongoose.model('Group', GroupSchema);
